Jenkinsfile

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env groovy
+
+@Library('sec_ci_libs@v2-latest') _
+
+def master_branches = ["master", ] as String[]
+
+pipeline {
+  agent {
+    dockerfile {
+      args  '--shm-size=2g'
+    }
+  }
+
+  environment {
+    JENKINS_VERSION = 'yes'
+  }
+
+  options {
+    timeout(time: 1, unit: 'HOURS')
+  }
+
+  stages {
+    stage('Authorization') {
+      steps {
+        user_is_authorized(master_branches, '8b793652-f26a-422f-a9ba-0d1e47eb9d89', '#frontend-dev')
+      }
+    }
+
+    stage('Initialization') {
+      steps {
+        ansiColor('xterm') {
+          retry(2) {
+            sh '''npm install'''
+          }
+        }
+      }
+    }
+
+    stage('Unit Test') {
+      steps {
+        ansiColor('xterm') {
+          sh '''npm run test'''
+        }
+      }
+    }
+
+    stage('Build') {
+      steps {
+        ansiColor('xterm') {
+          sh '''npm run dist'''
+        }
+      }
+
+      post {
+        always {
+          archiveArtifacts 'dist/*'
+        }
+      }
+    }
+
+    stage('Deploy'){
+      when {
```

Comment:
We should only deploy when a PR is merged so probably a separated hook. Seems like here will deploy every time there is a PR successful merged or not merged, right @DanielMSchmidt?

Reply:
Nope, the when clause with the expression ```env.BRANCH_NAME == "master"``` you highlighted here causes Jenkins to skip the step if it is run from a PR and not on master

Reply:
You can see it in action in the builds I triggered: 

[Here you can see the deploy is not done](https://jenkins.mesosphere.com/service/jenkins/blue/organizations/jenkins/frontend%2Fdcos-labs%2FUI-Kit%20Pipeline/detail/PR-5/2/pipeline/)
![bildschirmfoto 2018-03-27 um 09 11 36](https://user-images.githubusercontent.com/1337046/37952104-fa99e06c-319e-11e8-9ffc-02fd1dc9f5aa.png)



[Here you can see I commented the line out, so it's done](https://jenkins.mesosphere.com/service/jenkins/blue/organizations/jenkins/frontend%2Fdcos-labs%2FUI-Kit%20Pipeline/detail/PR-5/15/pipeline)
![bildschirmfoto 2018-03-27 um 09 11 39](https://user-images.githubusercontent.com/1337046/37952111-025ffb10-319f-11e8-920d-e2f4f32fb924.png)

Reply:
Awesome, thanks for explaining

Reply:
No worries 👍